### DISCUSSION TITLE
Table: using useControlledState to control selection

### DISCUSSION THREAD

**Discussion by cfranklyn** [iTwin/iTwinUI#1564] [Q&A] [answered]
Hello,

I'm running into an issue using the useControlledState property of the Table component. I think I may be using it incorrectly but I can't tell what I'm doing wrong.

Here's a sandbox showing what I'm trying to do:
https://codesandbox.io/s/gracious-sunset-cpy8qg?file=/src/TwoTablesComponent.tsx

I have two tables and I'm trying to clear the selection in one table when any items are selected in the other table, e.g.:
1) Select "Name 1" item.
2) Select "Name 4" item.
3) "Name 1" item is deselected

To do this, I am keeping track of the selection state outside of the table component and trying to update both tables when the selection state changes.

Everything is working as expected, but if you do the above steps, then select e.g. "Name 2", the "Name 1" item becomes selected again (same behavior for "Name 4" if you select "Name 5").

From what I can tell, when the onSelect method is run after "Name 2" is selected, it is sending both "Name 1" and "Name 2" (instead of just "Name 2").
So I must be missing some step where I am not really clearing the underlying Table selection state, but I can't tell what I'm leaving out. 

Does anyone know what I'm doing wrong here? Or what is the correct method for clearing selection from outside the table component?

**Comment by mayank99** [comment]
i can't reproduce this in your sandbox. could you verify and maybe provide a screen recording?

**Comment by cfranklyn** [comment]
Here it is:

![7DEF0586-76FA-416E-BCF1-27C547116019](https://github.com/iTwin/iTwinUI/assets/92109722/733e2534-ed38-449a-89b0-d7b639c2c62d)

If you watch the "Name 1" item, you can see what the issue is. It _appears_ to be deselected when the "Name 4" item is picked (which is correct), but in actuality, when the "Name 3" item is picked, it becomes selected again.

I guess what I'm looking for is a way to both keep track of items selected in the two tables from the parent component (I'm using onSelect to do this), _and_ to be able to programmatically clear the selection in a table (i.e. when an item is selected in another table).

**Reply by mayank99** [reply, marked as answer]
This is quite strange indeed. We're looking into it, but in the meantime here's a workaround using `key`. https://codesandbox.io/s/laughing-rain-hpnf3v?file=/src/TwoTablesComponent.tsx:2670-2722

Please let us know if `key` works in your real scenario so we can stop looking into it. The plan is to rewrite the Table in the coming months, so we'd like to avoid spending too much time on the current Table.

**Reply by cfranklyn** [reply]
Introducing this in my real scenario does fix the issue, but it introduces a couple of other selection problems (can't select more than two items at once, scrolling down the table and selecting an item forces a reset of the scrollbar to the top). 

But I can't reproduce either of these in the sample (which works just fine with your fix as far as I can tell), so I'm going to mark your answer as correct and assume that there is some difference in the way that I am setting up my tables in my real scenario and the way that I set them up in the sample. If I can find it, I'll reply to this thread.
